ipa-core/src/protocol/ipa_prf/quicksort.rs

```diff
@@ -115,18 +111,20 @@ where
 ///
 /// # Panics
 /// If you provide any invalid ranges, such as 0..0
-pub async fn quicksort_ranges_by_key_insecure<K, F, S>(
-    ctx: SemiHonestContext<'_>,
+pub async fn quicksort_ranges_by_key_insecure<C, K, F, S>(
+    ctx: C,
     list: &mut [S],
     desc: bool,
     get_key: F,
     mut ranges_to_sort: Vec<Range<usize>>,
 ) -> Result<(), Error>
 where
+    C: UpgradableContext,
```

Comment:
We will need this to implement malicious security for quicksort, because it needs to do intermediate validates and reveals.